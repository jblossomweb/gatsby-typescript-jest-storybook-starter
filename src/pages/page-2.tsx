import React from "react"
import { Link } from "gatsby"

import Template from "../templates/default"
import SEO from "../components/seo"

const SecondPage: React.FC = () => (
  <Template>
    <SEO title="Page two" />
    <h1>Hi from the second page</h1>
    <p>Welcome to page 2</p>
    <Link to="/">Go back to the homepage</Link>
  </Template>
)

export default SecondPage
