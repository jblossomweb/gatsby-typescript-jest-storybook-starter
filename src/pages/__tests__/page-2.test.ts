import snapshotTests from "../../../utils/story-snapshot-tests"

jest.mock("../../templates/default")
jest.mock("../../components/seo")

import { stories } from "./page-2.stories"

describe("pages/page-2", () => {
  snapshotTests(stories)
})
