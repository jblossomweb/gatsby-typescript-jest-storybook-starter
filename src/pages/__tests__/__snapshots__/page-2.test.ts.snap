// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/page-2 sample matches snapshot 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Test Title
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0 1.0875rem 1.45rem",
      }
    }
  >
    <main>
      <h1>
        Hi from the second page
      </h1>
      <p>
        Welcome to page 2
      </p>
      <a
        href="/"
      >
        Go back to the homepage
      </a>
    </main>
    <footer>
      Â© 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
        target="_blank"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
